twake/backend/node/src/services/documents/services/index.ts

```diff
@@ -514,18 +516,30 @@ export class DocumentsService {
         throw Error("Drive item doesn't exist");
       }
 
-      if (
-        !item.access_info.authorized_entities.find(entity =>
-          entityIdentityCheck(entity.id, context),
-        )
-      ) {
-        return false;
+      if (token) {
+        if (!item.access_info.public.token) {
+          return false;
+        }
+
+        const { token: itemToken, level: itemLevel } = item.access_info.public;
+
+        return itemLevel === level && itemToken === token;
```

Comment:
Careful here, implicitelly if you have write access you also have read access:
"manage" includes "write" that includes "read" that includes "none".
Better to have a little utility for that, you could simply do this:
```
manage: 30
write: 20
read: 10
none: 0

hasLevel(level, requested) => level >= requested
```